### DISCUSSION TITLE
Instance Spine Segmentation

### DISCUSSION THREAD

**Discussion by oguzcanbekar** [TorchIO-project/torchio#660] [General]
Hi there firstly I would like to be appreciate for your great work I hope one day I can also help to improve your code~~

My first question is I am tryoing to use my spine datas to instance segmentation, I have some nii files and nrrd files which I was segment manually. With the convenience provided by Torchio, I am able to create my dataset easily but I have question at that point, my datas have generally [512,512,600] dimensions but I would like to upload them to model by [512,512] dimensions. I mean for each epoch I would like to train by just 1 slice image or with batch more slices. I hope I could explain myself clearly until you answer I will also search but hopefully we can figure it out my all problems until I finish my all code. For this purpose I opened this discussion and I would like to ask my questions from here. Thank you very much in advance. 

Basically this is my first code too by Torchio:

def Dataset_load(base_path):

    dataset_dir = Path(base_path)

    images_dir = dataset_dir  / 'Img/'
    labels_dir = dataset_dir  / 'Mask/'

    image_paths = sorted(images_dir.glob('*.nii'))
    label_paths = sorted(labels_dir.glob('*.nrrd'))
    print(len(image_paths), len(label_paths))

    assert len(image_paths) == len(label_paths)

    CT = 'ct'
    MASK = 'mask'
    subjects = []
    for (image_path, label_path) in zip(image_paths, label_paths):
        subject = tio.Subject(
            CT = tio.ScalarImage(image_path),
            MASK = tio.LabelMap(label_path),
        )

        subjects.append(subject)
    dataset = tio.SubjectsDataset(subjects)
    return dataset

dataset = Dataset_load("DATASET")

**Comment by fepegar** [comment]
Hi, @oguzcanbekar. I suggest you read carefully the documentation on [Patch-based pipelines](https://torchio.readthedocs.io/data/patch_based.html). You could use the `Queue` and patches of size `(512, 512, 1)` to train with a 2D model.
